forte/data/data_store.py

```diff
@@ -533,9 +552,20 @@ def next_entry(self, tid: int) -> List:
             tid (int): Unique id of the entry.
 
         Returns:
-            The next entry of the same type as the ``tid`` entry.
+            A list of attributes representing the next entry of the same type
+            as the ``tid`` entry.
+
+        Raises:
+            IndexError: An error occurred accessing the next entry of the last
+                element in entry list.
         """
-        raise NotImplementedError
+        _, entry_type, index_id = self.get_entry(tid=tid)
+        entry_list = self.__elements[entry_type]
+        if index_id >= len(entry_list) - 1:
+            raise IndexError(
```

Comment:
if `index_id == len(entry_list) - 1`, we can return None instead of raising an exception since its still an expected input.

Reply:
I actually have a question about the index here. Why do we need to access the `entry_list` via `index_id + 1`?

Reply:
And this may help me understand why an out-of-bound index is expected.

Reply:
So we first find the `index_id` of the current entry with `tid`. Then we find the next entry, which has index `index_id+1`.

Reply:
Oh, forgot we are working on `next_entry` here, but why is `len(entry_list) - 1` valid but return None?

Reply:
I think since we may use next_entry to iterate through a list, we don't want to get an exception when reaching the end of the list. We only need `None` to indicate that this is the last one.

Reply:
Here are some related thoughts:

1. In Python the correct way to deal with iterator/loop end is to raise an iteration for `StopIteration`, what you proposed can be achieved by providing iterators of a certain entry type.
2. Consider we expand the behavior of `next_entry` with a parameter `step = 1` (which mimics the step parameter in list iteration), when should we use None?